samples/snippets/append_rows_proto2.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START bigquerystorage_append_rows_proto2]
+import datetime
+import decimal
+from typing import Iterable
+
+from google.cloud.bigquery_storage_v1beta2 import types
+from google.cloud.bigquery_storage_v1beta2.services import big_query_write
+from google.protobuf import descriptor_pb2
+
+# If you make updates to sample_data.proto, run:
+#
+#   protoc --python_out=. sample_data.proto
+#
+# to generate the sample_data_pb2 module.
+from . import sample_data_pb2
+
+
+def append_rows_proto2(project_id: str, dataset_id: str, table_id: str):
+    """Create a write stream, write some sample data, and commit the stream."""
+    write_client = big_query_write.BigQueryWriteClient()
+    parent = write_client.table_path(project_id, dataset_id, table_id)
+    write_stream = types.WriteStream()
+
+    # When creating the stream, choose the type. Use the PENDING type to wait
+    # until the stream is committed before it is visible. See:
+    # https://cloud.google.com/bigquery/docs/reference/storage/rpc/google.cloud.bigquery.storage.v1beta2#google.cloud.bigquery.storage.v1beta2.WriteStream.Type
+    write_stream.type_ = types.WriteStream.Type.PENDING
+    write_stream = write_client.create_write_stream(
+        parent=parent, write_stream=write_stream
+    )
+
+    # Some stream types support an unbounded number of requests. Pass a
+    # generator or other iterable to the append_rows method to continuously
+    # write rows to the stream as requests are generated.
```

Comment:
yeah, the caveat here is if you're not interleaving reads/writes you can end up in a case where no more requests are accepted until you receive responses.  I believe it's currently 1000, but that's an impl detail currently and not a formal part of the docs.

Reply:
Good point. I wonder if I should add a warning that it's highly recommended to read the responses and/or make the writes in a background thread? -- Plus I'll have to tell people not to use `multiprocessing` since gRPC hasn't really been playing nice with that.